### ISSUE TITLE
--root main as default?

### ISSUE THREAD

**Issue by sydow** [actonlang/acton#718]
Is it only me who lose a lot of time by forgetting to add --root main to the actonc command? You have a test program that you have already run so you have an executable, you make a change and recompile (without --root main), you run the program and become confused by the result. What would be the disadvantage of having --root main as default to 'actonc XXX.act'?

**Comment by plajjan** [comment]
@nordlander and I have discussed this previously and also agreed that it would be a sensible default to use `main` as the root actor, see #358. I did a stupid naive test of simply changing the `--root` arg, but that obviously doesn't work since it would start **requiring** modules to have a `main` actor. We need to check if there is a `main` actor and only then try building an executable out of it.

I've done some refactoring of the compiler and through that I tried to prepare for this. There is now a binTasks list for which we build executable binaries and the root actor name is qualified too, so we can have multiple items for which we want to build binaries in an acton project. All that's missing is to actually inspect the AST and find actors named `main`. @sydow you are more than welcome to implement this :)
